### DISCUSSION TITLE
ResponseParts::new to use Axum response extensions in custom tower service

### DISCUSSION THREAD

**Discussion by yaroslavros** [tokio-rs/axum#1385] [Q&A] [answered]
### Motivation

Today we have a very useful RequestParts::new to leverage Axum request extractors from tower services.
Would be great to have a similar capability in ResponseParts.

At the moment I am implementing an authentication layer and was looking to rely on axum-cookies. I can trivially extract CookieJar from RequestParts::new, but after processing authentication and awaiting for response, I cannot re-inject the modified CookieJar from the service layer.

I'm sure there are other potential IntoResponseParts applications that could be useful from intermediate layers.

### Proposal

I believe a trivial RequestParts::new method would address this use case.

### Alternatives

Potentially removing (crate) restriction on inner of ResponseParts would accomplish the same goal but probably would be less consistent with RequestParts.

**Comment by davidpdrsn** [comment, marked as answer]
You can do

```rust
(cookie_jar, response).into_response()
```

And return that from your `Service`.

**Comment by yaroslavros** [comment]
That worked.
Thank you very much for the quick response.
